drivers/storage/portworx/component/telemetry.go

```diff
@@ -59,16 +83,360 @@ func (t *telemetry) Delete(cluster *corev1.StorageCluster) error {
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind:     "Role",
+				Name:     CollectorRoleName,
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+		ownerRef,
+	)
+}
+
+func (t *telemetry) createProxyConfigMap(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	arcusLocation, present := cluster.Annotations[pxutil.AnnotationTelemetryArcusLocation]
+	if !present || len(arcusLocation) == 0 {
+		arcusLocation = "rest.cloud-support.purestorage.com"
+	}
+
+	config := fmt.Sprintf(
+		`
+admin:
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: 9901
+
+static_resources:
+  listeners:
+  - name: listener_cloud_support
+    address:
+      socket_address:
+        address: 127.0.0.1
+        port_value: 10000
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.http_connection_manager
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          stat_prefix: ingress_http
+          access_log:
+          - name: envoy.access_loggers.stdout
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.access_loggers.stream.v3.StdoutAccessLog
+          http_filters:
+          - name: envoy.filters.http.router
+          route_config:
+            name: local_route
+            virtual_hosts:
+            - name: local_service
+              domains: ["*"]
+              routes:
+              - match:
+                  prefix: "/"
+                request_headers_to_add:
+                - header:
+                   key: "product-name"
+                   value: "portworx"
+                - header:
+                   key: "appliance-id"
+                   value: "%s"
+                - header:
+                   key: "component-sn"
+                   value: "portworx-metrics-node"
+                - header:
+                   key: "product-version"
+                   value: "%s"
+                route:
+                  host_rewrite_literal: %s
+                  cluster: cluster_cloud_support
+  clusters:
+  - name: cluster_cloud_support
+    type: STRICT_DNS
+    dns_lookup_family: V4_ONLY
+    lb_policy: ROUND_ROBIN
+    load_assignment:
+      cluster_name: cluster_cloud_support
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: %s
+                port_value: 443
+    transport_socket:
+      name: envoy.transport_sockets.tls
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
+        common_tls_context:
+          tls_certificates:
+          - certificate_chain:
+              filename: /appliance-cert/cert
+            private_key:
+              filename: /appliance-cert/private_key
+`, cluster.UID, pxutil.GetPortworxVersion(cluster), arcusLocation, arcusLocation)
```

Comment:
when do we populate the Status, i am afraid we may run into race before the status is set.

Reply:
Your are right. What we can do it not create this component until cluster.Status.ClusterUID is not set. Any way, there is no point of starting the metrics collector until portworx is up.

Reply:
I did some test, if we wait for the UID and px does not come online, we will not have metrics for other components too. although we may not have autopilot metrics today but will have that eventually and more components will add metrics, the behavior is not good. do we have to use cluster UID, since it is just an ID maybe just use cluster name which is even more readable than a UUID, what do you think?